AppDB/test/unit/test_datastore_server.py

```diff
@@ -1007,5 +1010,71 @@ def test_apply_txn_changes(self):
 
     dd.apply_txn_changes(app, txn)
 
+  def test_delete_entities_txn(self):
+    app = 'guestbook'
+    txn_hash = {'root_key': 1}
+    txn_str = '1'.zfill(ID_KEY_LENGTH)
+    entity = self.get_new_entity_proto(app, *self.BASIC_ENTITY[1:])
+
+    db_batch = flexmock()
+    dd = DatastoreDistributed(db_batch, None)
+
+    keys = [entity.key()]
+    prefix = dd.get_table_prefix(entity.key())
+    entity_key = dd.get_entity_key(prefix, entity.key().path())
+    encoded_path = str(
+      dd._DatastoreDistributed__encode_index_pb(entity.key().path()))
+    txn_keys = [dd._SEPARATOR.join([app, txn_str, '', encoded_path])]
+    txn_values = {
+      txn_keys[0]: {
+        dbconstants.TRANSACTIONS_SCHEMA[0]: dbconstants.TxnActions.DELETE,
+        dbconstants.TRANSACTIONS_SCHEMA[1]: entity_key,
+        dbconstants.TRANSACTIONS_SCHEMA[2]: ''
+      }
+    }
+
+    flexmock(dd).should_receive('get_root_key').and_return('root_key')
+
+    db_batch.should_receive('batch_put_entity').with_args(
+      dbconstants.TRANSACTIONS_TABLE,
+      txn_keys,
+      dbconstants.TRANSACTIONS_SCHEMA,
+      txn_values,
+      ttl=dd.MAX_TXN_DURATION * 2
+    )
+    dd.delete_entities_txn(app, keys, txn_hash)
+
+  def test_put_entities_txn(self):
+    app = 'guestbook'
```

Comment:
The properties are defined in `self.BASIC_ENTITY`. I just needed the app ID for the transaction key.